crates/ruff/src/rules/refurb/rules/repeated_append.rs

```diff
@@ -0,0 +1,442 @@
+        return;
+    };
+
+    // No need to proceed if we have less than 1 `append` to work with.
+    if appends.len() <= 1 {
+        return;
+    }
+
+    // group borrows from checker, so we can't directly push into checker.diagnostics
+    let diagnostics: Vec<Diagnostic> = group_appends(appends)
+        .iter()
+        .filter_map(|group| {
+            // Groups with just one element are fine, and shouldn't be replaced by `extend`.
+            if group.appends.len() <= 1 {
+                return None;
+            }
+
+            let replacement = make_suggestion(group, checker.generator());
+
+            let mut diagnostic = Diagnostic::new(
+                RepeatedAppend {
+                    name: group.name().to_string(),
+                    replacement: SourceCodeSnippet::new(replacement.clone()),
+                },
+                group.range(),
+            );
+
+            // We only suggest a fix when all appends in a group are clumped together. If they're
+            // non-consecutive, fixing them is much more difficult.
+            if checker.patch(diagnostic.kind.rule()) && group.is_consecutive {
+                diagnostic.set_fix(Fix::suggested(Edit::replacement(
+                    replacement,
+                    group.start(),
+                    group.end(),
+                )));
+            }
+
+            Some(diagnostic)
+        })
+        .collect();
+
+    checker.diagnostics.extend(diagnostics);
+}
+
+#[derive(Debug, Clone)]
+struct Append<'a> {
+    /// Receiver of the `append` call (aka `self` argument).
+    receiver: &'a ast::ExprName,
+    /// [`BindingId`] that the receiver references.
+    binding_id: BindingId,
+    /// [`Binding`] that the receiver references.
+    binding: &'a Binding<'a>,
+    /// [`Expr`] serving as a sole argument to `append`.
+    argument: &'a Expr,
+    /// The statement containing the `append` call.
+    stmt: &'a Stmt,
+}
+
+#[derive(Debug)]
+struct AppendGroup<'a> {
+    /// A sequence of `appends` connected to the same binding.
+    appends: Vec<Append<'a>>,
+    /// `true` when all appends in the group follow one another and don't have other statements in
+    /// between. It is much easier to make fix suggestions for consecutive groups.
+    is_consecutive: bool,
+}
+
+impl AppendGroup<'_> {
+    fn name(&self) -> &str {
+        assert!(!self.appends.is_empty());
+        &self.appends.first().unwrap().receiver.id
+    }
+}
+
+impl Ranged for AppendGroup<'_> {
+    fn range(&self) -> TextRange {
+        assert!(!self.appends.is_empty());
+        TextRange::new(
+            self.appends.first().unwrap().stmt.start(),
+            self.appends.last().unwrap().stmt.end(),
+        )
+    }
+}
+
+/// Match consecutive calls to `append` on list variables starting from the given statement.
+fn match_consecutive_appends<'a>(
+    semantic: &'a SemanticModel,
+    stmt: &'a Stmt,
+) -> Option<Vec<Append<'a>>> {
```

Comment:
@SavchenkoValeriy - I ended up reverting to `Option`, your initial version was better :joy: